### DISCUSSION TITLE
NoTracking not compatible with patching

### DISCUSSION THREAD

**Discussion by loraderon** [ravendb/ravendb#19297] [Q&A] [answered]
When setting `NoTracking` to true I understand it's impossible to call `session.Store(Async)` and that is a good thing.

However, why is the same check made in `session.SaveChanges(Async)`? That makes it impossible to use the same session for patching untracked entities. Is this by design or an oversight?

**Comment by ayende** [comment, marked as answer]
If you have `NoTracking` - that means that no changes are tracked. But it is easy to lose track of that.
The error is there as a way to avoid mistakes - because you set `NoTracking` somewhere, loaded an entity and modified it, then tried to save it.

Since you are not tracking anything, if you want to do patching, using a separate session
